frontend/src/components/InformationDrawer/InformationDrawer.js

```diff
@@ -0,0 +1,110 @@
+import React, { useState, useCallback, useRef } from 'react';
+import { debounce } from 'lodash';
+import {
+  List,
+  ListItem,
+  ListItemIcon,
+  ListItemText,
+  SwipeableDrawer,
+  TextField,
+  Typography,
+} from '@material-ui/core';
+import { SdCard, Schedule, Movie, Crop, Close } from '@material-ui/icons';
+
+export default function InformationDrawer({
+  open,
+  onOpen,
+  onClose,
+  videoTitle = '',
+  videoDuration = '',
+  videoSize = '',
+  videoResolution = '',
+  videoFormat = '',
+  onVideoTitleChange,
+}) {
+  const [title, setTitle] = useState(videoTitle);
+
+  const handleTitleChange = (newTitle) => {
+    onVideoTitleChange(newTitle);
+  };
+
+  const debouncedHandleTitleChange = useCallback(
+    debounce(handleTitleChange, 300),
+    [],
+  );
+
+  const delayedHandleTitleChange = (event) => {
+    const newTitle = event.target.value;
+    setTitle(newTitle);
+
+    debouncedHandleTitleChange.cancel();
+    debouncedHandleTitleChange(newTitle);
+  };
+
+  const videoInformation = useRef(
+    new Map([
+      [videoDuration, { icon: <Schedule /> }],
+      [videoSize, { icon: <SdCard /> }],
+      [videoResolution, { icon: <Crop /> }],
+      [videoFormat, { icon: <Movie /> }],
+    ]),
+  );
+
+  const renderSidebarTitle = () => (
+    <ListItem>
+      <ListItemIcon
+        style={title.length === 0 ? null : { cursor: 'pointer' }}
```

Comment:
Why are you changing the style here? If you want to make it impossible to close the drawer if title is empty, then why not just disable the icon? if `ListItemIcon` does not have disabled property, then use `IconButton` here(as a child component of `ListItemIcon` or completely replace it).

Reply:
Because even with disabled it remains a cursor pointer